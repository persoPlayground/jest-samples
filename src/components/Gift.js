import React, { Component } from 'react';
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';
export default class Gift extends Component {

    state = { 
        person:'',
        gift:''
    }

    handleChange = e => {
        this.setState({ person: e.target.value });
    }
    handleChangeGift = e => {
        this.setState({gift: e.target.value})
    }
    render(){
        return( 

            <div className="gift">
                <Form>

                    <FormGroup>
                    <ControlLabel>Person</ControlLabel>
                    <FormControl
                            type="text"
                            className="input-person"
                            value={this.state.person}
                            placeholder="Person"
                            onChange={this.handleChange}
                        />
                    </FormGroup>    

    <FormGroup>
                 <ControlLabel>Gift</ControlLabel>
                 <FormControl
          type="text"
          className="input-gift"
          value={this.state.gift}
          placeholder="Gift"
          onChange={this.handleChangeGift}
          />
                    </FormGroup>    
                </Form>
                <Button className="btn-remove" 
                        onClick={ () => this.props.removeGift(this.props.gift.id)}>
                                Delete
                         </Button>   
         
            </div>
        )
    }
}