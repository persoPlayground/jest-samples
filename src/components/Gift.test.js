import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift';

describe('Gift', () => {

const mockRemove = jest.fn();
const id = 1;
//props interface
const props = { gift: {id}, removeGift: mockRemove };

const gift = shallow(<Gift {...props} />);
it('renders correctly', () =>{
   expect(gift).toMatchSnapshot();
});

it('Initilizes a person and gift in `state`', () => {
    expect(gift.state()).toEqual({ person:'', gift:''});
});

describe('when typing in person Input', () => {

    const person =  'Mo';
    beforeEach(() => {
        gift.find('.input-person').simulate('change', { target: { value: person } });
    })
    it('update `state` Person', () =>{
        expect(gift.state().person).toEqual(person);
    });
});


describe('when typing in Gift Input', () => {

    const newGift = 'blabla';
    beforeEach(() => {
        gift.find('.input-gift').simulate('change', { target: { value: newGift}});
    })
    it('updates `state` gift', () => {
        expect(gift.state().gift).toEqual(newGift);
    });
});

describe('when click remove button', () => {

    beforeEach(() =>{
        gift.find('.btn-remove').simulate('click');
    });
    it('calls the removGift callback', () => {
 //mock function: 
        expect(mockRemove).toHaveBeenCalledWith(id);
    });
})
});