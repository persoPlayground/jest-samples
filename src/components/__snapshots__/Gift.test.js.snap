// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
        Object {
            "id": 1,
          }
    }
    removeGift={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
>
          <FormGroup
                    bsClass="form-group"
          >
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}
                    >
                              Person
                    </ControlLabel>
                    <FormControl
                              bsClass="form-control"
                              className="input-person"
                              componentClass="input"
                              onChange={[Function]}
                              placeholder="Person"
                              type="text"
                              value=""
                    />
          </FormGroup>
          <FormGroup
                    bsClass="form-group"
          >
                    <ControlLabel
                              bsClass="control-label"
                              srOnly={false}
                    >
                              Gift
                    </ControlLabel>
                    <FormControl
                              bsClass="form-control"
                              className="input-gift"
                              componentClass="input"
                              onChange={[Function]}
                              placeholder="Gift"
                              type="text"
                              value=""
                    />
          </FormGroup>
</Form>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn-remove"
          disabled={false}
          onClick={[Function]}
>
          Delete
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <FormGroup
              bsClass="form-group"
>
              <ControlLabel
                            bsClass="control-label"
                            srOnly={false}
              >
                            Person
              </ControlLabel>
              <FormControl
                            bsClass="form-control"
                            className="input-person"
                            componentClass="input"
                            onChange={[Function]}
                            placeholder="Person"
                            type="text"
                            value=""
              />
</FormGroup>,
            <FormGroup
              bsClass="form-group"
>
              <ControlLabel
                            bsClass="control-label"
                            srOnly={false}
              >
                            Gift
              </ControlLabel>
              <FormControl
                            bsClass="form-control"
                            className="input-gift"
                            componentClass="input"
                            onChange={[Function]}
                            placeholder="Gift"
                            type="text"
                            value=""
              />
</FormGroup>,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
>
                  Person
</ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-person"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Person"
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Person",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Person",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-person",
                  "componentClass": "input",
                  "onChange": [Function],
                  "placeholder": "Person",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
>
                  Gift
</ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  className="input-gift"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Gift"
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Gift",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Gift",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "className": "input-gift",
                  "componentClass": "input",
                  "onChange": [Function],
                  "placeholder": "Gift",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Delete",
          "className": "btn-remove",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
>
            <FormGroup
                        bsClass="form-group"
            >
                        <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                        >
                                    Person
                        </ControlLabel>
                        <FormControl
                                    bsClass="form-control"
                                    className="input-person"
                                    componentClass="input"
                                    onChange={[Function]}
                                    placeholder="Person"
                                    type="text"
                                    value=""
                        />
            </FormGroup>
            <FormGroup
                        bsClass="form-group"
            >
                        <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                        >
                                    Gift
                        </ControlLabel>
                        <FormControl
                                    bsClass="form-control"
                                    className="input-gift"
                                    componentClass="input"
                                    onChange={[Function]}
                                    placeholder="Gift"
                                    type="text"
                                    value=""
                        />
            </FormGroup>
</Form>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn-remove"
            disabled={false}
            onClick={[Function]}
>
            Delete
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <FormGroup
                bsClass="form-group"
>
                <ControlLabel
                                bsClass="control-label"
                                srOnly={false}
                >
                                Person
                </ControlLabel>
                <FormControl
                                bsClass="form-control"
                                className="input-person"
                                componentClass="input"
                                onChange={[Function]}
                                placeholder="Person"
                                type="text"
                                value=""
                />
</FormGroup>,
              <FormGroup
                bsClass="form-group"
>
                <ControlLabel
                                bsClass="control-label"
                                srOnly={false}
                >
                                Gift
                </ControlLabel>
                <FormControl
                                bsClass="form-control"
                                className="input-gift"
                                componentClass="input"
                                onChange={[Function]}
                                placeholder="Gift"
                                type="text"
                                value=""
                />
</FormGroup>,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
>
                    Person
</ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Person"
                    type="text"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Person",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Person",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-person",
                    "componentClass": "input",
                    "onChange": [Function],
                    "placeholder": "Person",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <ControlLabel
                    bsClass="control-label"
                    srOnly={false}
>
                    Gift
</ControlLabel>,
                  <FormControl
                    bsClass="form-control"
                    className="input-gift"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Gift"
                    type="text"
                    value=""
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "control-label",
                    "children": "Gift",
                    "srOnly": false,
                  },
                  "ref": null,
                  "rendered": "Gift",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "className": "input-gift",
                    "componentClass": "input",
                    "onChange": [Function],
                    "placeholder": "Gift",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Delete",
            "className": "btn-remove",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
